pinot-query-runtime/src/main/java/org/apache/pinot/query/service/QueryDispatcher.java

```diff
@@ -84,9 +87,10 @@ public ResultTable submitAndReduce(long requestId, QueryPlan queryPlan,
     return resultTable;
   }
 
-  public int submit(long requestId, QueryPlan queryPlan, long timeoutMs, Map<String, String> queryOptions)
+  public int submit(long requestId, QueryPlan queryPlan, long deadlineNanos, Map<String, String> queryOptions)
       throws Exception {
     int reduceStageId = -1;
+    Deadline deadline = Deadline.after(deadlineNanos - System.nanoTime(), TimeUnit.NANOSECONDS);
```

Comment:
could we put query deadline as not part of the GRPC dispatch request deadline but instead in the `metadata`?

IMO the deadline for the dispatch GRPC call is not related to the mailbox GRPC call or the query execution, as they return immediately after the plan is compiled. we might later on provide a different deadline for dispatch for the compilation.

Reply:
> we might later on provide a different deadline for dispatch for the compilation.

But I feel this should be subsumed right as part of the overall query deadline. I mean what if in a pathological scenario, dispatch for the compilation itself exhausts

Reply:
yes then it should failed faster as the compilation shouldn't take too long comparing to the actual execution.

Reply:
it is easier to use single deadline for all requests. I can do a follow up as optimization to set the plan rpc deadline shorter

Reply:
Sounds good in this case plz add a comment in the QueryServer to explain the deadline is not only used for dispatch grpc but also for entire l query

Reply:
done